tools/evict-shared-cache/evict-shared-cache.cpp

```diff
@@ -0,0 +1,109 @@
+/*
+ * Copyright 2023 SiFive, Inc.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You should have received a copy of LICENSE.Apache2 along with
+ * this software. If not, you may obtain a copy at
+ *
+ *    https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+// Open Group Base Specifications Issue 7
+#define _XOPEN_SOURCE 700
+#define _POSIX_C_SOURCE 200809L
+
+#include <string.h>
+
+#include <iostream>
+#include <memory>
+
+#include "eviction-policy.h"
+#include "gopt/gopt-arg.h"
+#include "gopt/gopt.h"
+
+void print_help(const char* argv0) {
+  // clang-format off
+  std::cout << std::endl
```

Comment:
I'd make this `cerr` instead and I'd exit unconditionally.

Reply:
I didn't put the exit in here because sometimes I want `exit(EXIT_SUCCESS)` and sometimes I want `exit(EXIT_FAILURE)`

Reply:
I never thought it was bad to exit successfully when asking for help personally but I can accept that argument. The main issue with using `cout` is that it will be pipped back to the process where as we can choose to log cerr instead which would be a better place to be notified that we messed up the call.

Reply:
Sure, I  `EXIT_SUCCESS`  when the user does `--help` but when they do something wrong like

`./bin/evict --policy`

I print that the policy needs a value, call `print_help` then `exit(FAIL)`